.github/workflows/build-check.yml

```diff
@@ -0,0 +1,18 @@
+name: Build check
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  push:
+    branches:
+      - main
+
+jobs:
+  compile_tests:
+    name: Compile tests
+    runs-on: ubuntu-latest
```

Comment:
Forgot about this. We have code that is specific to Windows. This will skip those files. Might need to add a matrix and include ubuntu, mac and windows

Reply:
Or at least linux and windows. Don't think we have mac tests

Reply:
thank you

Reply:
Good news is that I made the matrix. Bad news is that windows doesn't build